waiter/src/waiter/scheduler.clj

```diff
@@ -1251,3 +1212,121 @@
+  [clock timer-chan service-id->source-tokens-entries-fn service-id->metrics-fn retrieve-descriptor-fn
+   fallback-state-atom kv-store start-new-service-fn]
+  (cid/with-correlation-id
+    "start-new-services-maintainer"
+    (let [correlation-id (cid/get-correlation-id)
+          exit-chan (async/promise-chan)
+          query-chan (async/chan)
+          state-atom (atom {:last-update-time nil
+                            :service-id->last-request-time {}})
+          query-state-fn (fn query-state-fn [_]
+                           (assoc @state-atom :supported-include-params []))
+          go-chan
+          (async/go
+            (try
+              (loop [{:keys [service-id->last-request-time] :as current-state} @state-atom]
+                (reset! state-atom current-state)
+                (let [[msg current-chan]
+                      (async/alts! [exit-chan timer-chan query-chan] :priority true)
+                      next-state
+                      (condp = current-chan
+                        exit-chan
+                        (do
+                          (log/warn "stopping start-new-services-maintainer")
+                          (when (not= :exit msg)
+                            (throw (ex-info "Stopping start-new-services-maintainer" {:time (clock) :reason msg}))))
+
+                        timer-chan
+                        (timers/start-stop-time!
+                          (metrics/waiter-timer "core" "start-new-services-maintainer" "process-services")
+                          (try
+                            (let [count-atom (atom 0)
+                                  fallback-state @fallback-state-atom
+                                  new-service-id->last-request-time
+                                  (->> (service-id->metrics-fn)
+                                    seq
+                                    (map
+                                      (fn [[service-id {:strs [last-request-time]}]]
+                                        [service-id last-request-time]))
+                                    (filter
+                                      (fn [[_ last-request-time]]
+                                        (some? last-request-time)))
+                                    (into {}))
+                                  service-ids-with-new-last-request-times
+                                  (->> (seq new-service-id->last-request-time)
+                                    (filter
+                                      (fn [[service-id new-last-request-time]]
+                                        (let [old-last-request-time (get service-id->last-request-time service-id)]
+                                          (or (nil? old-last-request-time)
+                                              (t/after? new-last-request-time old-last-request-time)))))
+                                    (map first))
+                                  tokens-with-new-last-request-times
```

Comment:
Feels like we should query if a service is stale and receiving requests past the time it went stale. That should be the trigger to determine the token and start a new version of the service. It should work for any service including run-as-requester and parameterized.